### PR TITLE
Feature/restructure account type

### PR DESCRIPTION
**Pull request by gombaniro**

### CODE REVIEW

File: src/main/java/com/tg5/service/RoleServiceImpl.java [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
+import com.tg5.repository.AccountTypeRepository;
 import com.tg5.repository.RoleRepository;
 import com.tg5.service.contract.RolePayload;
 import edu.miu.common.service.BaseReadWriteServiceImpl;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+
 @Service
```
Comment (vytaux) [inline src/main/java/com/tg5/service/RoleServiceImpl.java:15]: Could do RequiredArgsConstructor and remove constructor.

File: src/main/java/com/tg5/service/AccountTypeServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class AccountTypeServiceImpl extends BaseReadWriteServiceImpl<AccountTypePayload, AccountType, Long> implements AccountTypeService {
+
+    private final AccountTypeRepository accountTypeRepository;
+
+    @Autowired
```
Comment (vytaux) [inline src/main/java/com/tg5/service/AccountTypeServiceImpl.java]: could do wihtout constructor with RequiredArgsConstructor

File: src/main/java/com/tg5/domain/AccountCurrency.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public enum MoneyType {VirtualDollar, Points}
+    private  BigDecimal balance;
+
+    @Enumerated(EnumType.STRING)
+    private  MoneyType moneyType;
+
+
+    public  AccountCurrency plus(AccountCurrency currency){
```
Comment (vytaux) [inline src/main/java/com/tg5/domain/AccountCurrency.java:31]: If this class not meant to be used directly, make this abstract?

**Comment by dip06ece** [comment]
I am okay with this change. Looks good!

**Review by dip06ece — APPROVED** [review]

File: src/main/java/com/tg5/service/contract/AccountTypePayload.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+@Data
+public class AccountTypePayload implements Serializable {
+    private static final long serialVersionUID = 4L;
+
+    private Long id;
+
+    private String name;
+
+    private String description;
+
```
Comment (vytaux) [inline src/main/java/com/tg5/service/contract/AccountTypePayload.java:16]: Could this just be balance? So we could update balance on PUT as well.

File: src/main/java/com/tg5/service/RoleServiceImpl.java [hunk 5]
```diff
@@ -1,30 +1,38 @@
...
     public RolePayload create(RolePayload request) {
-//        List<AccountType> types =  accountTypeRepository.findAccountTypesByIdIn(request.getDefaultAccountTypes());
-//        Role role = new Role();
-//        role.setName(request.getName());
-//        role.setDefaultAccountTypes(types);
-//        roleRepository.save(role);
-//        request.setId(role.getId());
+        List<AccountType> types =  accountTypeRepository.findAccountTypesByIdIn(request.getDefaultAccountTypes());
+        Role role = new Role();
+        role.setName(request.getName());
```
Comment (vytaux) [inline src/main/java/com/tg5/service/RoleServiceImpl.java:27]: would you need this create method, if you sent {"defaultAccountTypes": [{"id": 1}, {"id": 2}]}?

File: src/main/java/com/tg5/domain/AccountCurrency.java (hunk 3 above)
Comment (gombaniro) [reply src/main/java/com/tg5/domain/AccountCurrency.java:31]: that was a previous idea, but unable to make it hibernate recognized when used as a type in other classes,  I reverted to a concrete type. I haven't figured out how to make abstract work with hibernate

**Pull request merged**